replays/replays.lib.php

```diff
@@ -77,17 +77,50 @@ function toID($name) {
 		return preg_replace('/[^a-z0-9]+/','',$name);
 	}
 
-	function search($term, $page = 0, $isPrivate) {
+	function search($args) {
+		$page = isset(args["page"]) ? args["page"] : 0;
+
 		if (!$this->db) return [];
 		if ($page > 100) return [];
 
 		$limit1 = intval(50*($page-1));
 		if ($limit1 < 0) $limit1 = 0;
-		$term = $this->toID($term);
-		$isPrivate = $isPrivate ? 1 : 0;
-		// $res = $this->db->prepare("SELECT uploadtime, id, format, p1, p2 FROM ps_replays WHERE private = 0 AND (p1id = ? OR p2id = ?) ORDER BY uploadtime DESC LIMIT ?, 51");
-		$res = $this->db->prepare("(SELECT uploadtime, id, format, p1, p2, password FROM ps_replays FORCE INDEX (p1) WHERE private = ? AND p1id = ? ORDER BY uploadtime DESC) UNION (SELECT uploadtime, id, format, p1, p2, password FROM ps_replays FORCE INDEX (p2) WHERE private = ? AND p2id = ? ORDER BY uploadtime DESC) ORDER BY uploadtime DESC LIMIT ?, 51;");
-		$res->execute([$isPrivate, $term, $isPrivate, $term, $limit1]);
+
+		$isPrivate = (isset($args["isPrivate"]) && args["isPrivate"]) ? 1 : 0;
```

Comment:
So, in case it was unclear, `a ?? b` is equivalent to `isset(a) ? a : b`. Like `isset`, it bypasses the "variable not set" error.

```suggestion
		$isPrivate = ($args["isPrivate"] ?? null) ? 1 : 0;
```